### DISCUSSION TITLE
`closestTo` value is show in terminal but could not be utilized in the program

### DISCUSSION THREAD

**Discussion by emadbaqeri** [js-fns/date-fns#3016] [Q&A] [answered]
Hey folks, I'm trying to find the closest date based on today. I have used `closestTo` utility function. While I'm trying to log the output value of this function everything works fine and I see the output but since I'm using it in the program I get undefined out of it. Any Ideas would be appreciated :)

```javascript
import * as dateFns from 'date-fns';

const today = new Date().getTime();
const dates = [
  2022-04-10T14:07:12.276Z,
  2022-04-10T14:07:06.967Z,
  2022-04-10T14:07:04.663Z,
  2022-04-10T14:07:03.040Z,
  2022-04-10T14:07:01.420Z,
  2022-04-10T14:06:59.869Z,
  2022-04-10T14:06:53.223Z
]

const closestIndex = dateFns.closestTo(today, dates);

console.log(closestIndex); // => 0

console.log(dates[closestIndex]); // => undefined could not be used as index value
```

**Comment by fturmel** [comment, marked as answer]
Hi @emadbaqeri,

Please refer to the docs to see the proper function usage.

1. `closestTo` returns a date, while `closestIndexTo` returns an index.
2. The array passed need to be of JS Date instances or timestamps (ECMAScript epoch), not strings.
3. If the value of the returned index is `undefined`, it's likely because the array of dates passed to the function was empty.

https://date-fns.org/docs/closestTo
https://date-fns.org/docs/closestIndexTo

```js
import { closestIndexTo } from "date-fns";

const today = new Date();
const dates = [
  new Date("2022-04-10T14:07:12.276Z"),
  new Date("2022-04-10T14:07:06.967Z"),
  new Date("2022-04-10T14:07:04.663Z"),
  new Date("2022-04-10T14:07:03.040Z"),
  new Date("2022-04-10T14:07:01.420Z"),
  new Date("2022-04-10T14:06:59.869Z"),
  new Date("2022-04-10T14:06:53.223Z")
];

const closestIndex = closestIndexTo(today, dates);

console.log(closestIndex); // 0
console.log(dates[closestIndex].toISOString()); // 2022-04-10T14:07:12.276Z

console.log(closestIndexTo(today, [])); // undefined
```

**Reply by emadbaqeri** [reply]
Thanks @fturmel, the issue has been resolved.
By the way thanks for time ;)
